arangod/Aql/MaterializeExecutor.cpp

```diff
@@ -104,9 +110,14 @@ std::pair<ExecutionState, NoStats> arangodb::aql::MaterializeExecutor::produceRo
       TRI_ASSERT(state == ExecutionState::DONE);
       return {state, NoStats{}};
     }
-    auto collection =
-      reinterpret_cast<arangodb::LogicalCollection const*>(
-        input.getValue(colRegId).slice().getUInt());
+    arangodb::LogicalCollection const* collection = nullptr;
+    if constexpr (std::is_same<T, std::string const&>::value) {
+      collection = trx->documentCollection(collectionSource);
```

Comment:
Maybe store retrieved pointer as member variable between calls to produceRows  in this code branch. Looks like collection will be unchanged. Am  I missing something ?